pro enable landscape docs (#2674)
canonical/ubuntu-pro-client: docs/howtoguides/enable_landscape.rst

How to enable Landscape
***********************

``pro`` offers a convenient way to register a machine with Landscape via the

> ``pro enable landscape`` command.

Reviewer: `pro` offers convenient ways of accessing all the Pro services ;) can we remove that bit and just leave "you can register a machine with Landscape via the ``pro enable landscape`` command. Then, let's add a bit about the two different options (interactive/non-interactive) for enabling so that we can add a couple of the keywords that might lead people to find the page if they're searching for it (like hands off/automatic, that sort of thing).

Reviewer: Probably also an idea to state up front what info they'll need to complete the registration (which I assume is the same info no matter which method they choose?)

Reviewer: Another question that users might have (which occurred to me after I saw the 40k machines thing added to the index) - how do I register more than a few machines at once? Doing this one by one for an entire fleet would be quite inefficient so is there some method we recommend for doing this at larger scale?

Author: > `pro` offers convenient ways of accessing all the Pro services ;) can we remove that bit and just leave "you can register a machine with Landscape via the `pro enable landscape` command. Then, let's add a bit about the two different options (interactive/non-interactive) for enabling so that we can add a couple of the keywords that might lead people to find the page if they're searching for it (like hands off/automatic, that sort of thing).

Sounds good, I'll remove/add these.

> Probably also an idea to state up front what info they'll need to complete the registration (which I assume is the same info no matter which method they choose?)

Good idea, I'll add it.

> how do I register more than a few machines at once?

This is out of scope for pro-client. The correct answer is dependent on the customer's infrastructure setup. We can add a line about the non-interactive case being useful for automation, but it is up to the customer to run the command on many machines.

Enable interactively
====================

To register your machine by interactively providing your Landscape account
details at the CLI, run:

.. code:: bash

   sudo pro enable landscape

> That command will install ``landscape-client`` and drop you into an interactive

Author: Yep I'll change it to that wording

wizard to complete the Landscape registration for the machine.

Enable non-interactively
========================

If you know the details of your Landscape setup and would like to register a

> machine without any interaction necessary, you can pass ``landscape-config``

Reviewer: ```suggestion
machine without using the wizard, you can pass ``landscape-config``
```

Reviewer: Also, a question a reader might naturally have is "what's landscape-config and do I need to install it?". It's probably worth mentioning that it's part of the pro enable landscape command.

Author: I'll make the suggested change.

`landscape-config` isn't part of `pro enable landscape` per se. It is a command installed as part of the `landscape-client` package and owned by the landscape team. It could completely change underneath of us without us knowing, so we shouldn't try to document it too much. The only web-accessible documentation that I can find (which I would've linked before but didn't find it until now), is the manpage on line here: https://manpages.ubuntu.com/landscape-config

So I'll add that link and an explanation that under-the-hood, pro enable landscape installs landscape-config and executes it, which should answer the "do I need to install it" part

parameters to ``pro enable landscape`` after a ``--``. You should also use the

> ``--assume-yes`` flag to ensure there are no prompts.

Reviewer: One question I have (and is probably not really a problem for us but still might occur to a reader) is - if we can and should avoid the prompts by using --assume-yes, why are the prompts there by default? Surely it should have been that yes is assumed by default and that you can turn that off by doing --assume-no. Can you --assume-no? What is assume-yes actually doing? (I assume it's not just suppressing prompts, but is rather doing something else - I think it's better to be clear here to avoid raising questions). (also sorry, that's way more than one question)

Author: We only _should_ avoid prompts if you want a completely non-interactive enablement. The prompts are there in the `landscape-config` command that we don't own/control. 

Prompts can be useful - if you provide some but not all required parameters to landscape-config, it will prompt you for the other required fields. If you miss a required parameter with `--assume-yes` then it will just fail. The "interactive but with some parameters supplied" is a valid invocation of the command, just not something I think is worth recommending - so I didn't document it here.

The "problem" is that `landscape-config` currently prompts for non-required parameters, and that is why we _have_ to recommend `--assume-yes` here. That is arguably a UX problem with `landscape-config` but that is out of our control (and the landscape team is aware of it).

What `--assume-yes` is really doing is accepting defaults for those non-required parameters - that should probably be the description instead of "ensure there are no prompts".

Reviewer: Understood, thanks for the context there. I think that given this situation, the edit you made is the best one that could be made. It neatly sidesteps the entire thorny issue - good job :)

> Altogether, it should look something like this:
> 
> .. code:: bash
> 
>    sudo pro enable landscape --assume-yes -- --account-name my-account --computer-title my-computer

Reviewer: ```suggestion
The command to enable Landscape takes the following format:

.. code:: bash

   sudo pro enable landscape -- <landscape-config parameters>

Which, when the parameters are added, should look something like this:

.. code:: bash

   sudo pro enable landscape --assume-yes -- --account-name <my-account> --computer-title <my-computer>
```

Reviewer: Also, do we need to list all the (currently) possible parameters?

Author: I'll make that change, and I posted above a link that I'll include which describes the possible parameters

That command will install ``landscape-client`` and pass the provided parameters
after ``--`` to the ``landscape-config`` tool to automatically register the
machine.

What next?
==========

After successfully running ``pro enable landscape``, either interactively
or non-interactively, an administrator of your Landscape account will need to go

> to the Pending Computers page in Landscape to accept the machine you just

Reviewer: ```suggestion
to the "Pending Computers" page in Landscape to accept the machine you just
```

registered.

> And that's it! The machine should now show up in the Landscape dashboard for

Reviewer: ```suggestion
And that's it! The machine should now appear in the Landscape dashboard for
```
